processor/src/test/java/org/mapstruct/ap/internal/util/StringsTest.java

```diff
@@ -0,0 +1,99 @@
+        assertThat( Strings.capitalize( "capitalize" ) ).isEqualTo( "Capitalize" );
+        assertThat( Strings.capitalize( "AlreadyCapitalized" ) ).isEqualTo( "AlreadyCapitalized" );
+        assertThat( Strings.capitalize( "notCapitalized" ) ).isEqualTo( "NotCapitalized" );
+    }
+
+    @Test
+    public void testDecapitalize() throws Exception {
+        assertThat( Strings.decapitalize( null ) ).isNull();
+        assertThat( Strings.decapitalize( "c" ) ).isEqualTo( "c" );
+        assertThat( Strings.decapitalize( "capitalize" ) ).isEqualTo( "capitalize" );
+        assertThat( Strings.decapitalize( "AlreadyCapitalized" ) ).isEqualTo( "alreadyCapitalized" );
+        assertThat( Strings.decapitalize( "notCapitalized" ) ).isEqualTo( "notCapitalized" );
+    }
+
+    @Test
+    public void testJoin() throws Exception {
+        assertThat( Strings.join( new ArrayList<String>(), "-" ) ).isEqualTo( "" );
+        assertThat( Strings.join( Arrays.asList( "Hello", "World" ), "-" ) ).isEqualTo( "Hello-World" );
+        assertThat( Strings.join( Arrays.asList( "Hello" ), "-" ) ).isEqualTo( "Hello" );
+    }
+
+    @Test
+    public void testJoinAndCamelize() throws Exception {
+        assertThat( Strings.joinAndCamelize( new ArrayList<String>() ) ).isEqualTo( "" );
+        assertThat( Strings.joinAndCamelize( Arrays.asList( "Hello", "World" ) ) ).isEqualTo( "HelloWorld" );
+        assertThat( Strings.joinAndCamelize( Arrays.asList( "Hello", "world" ) ) ).isEqualTo( "HelloWorld" );
+        assertThat( Strings.joinAndCamelize( Arrays.asList( "hello", "world" ) ) ).isEqualTo( "helloWorld" );
+    }
+
+    @Test
+    public void testIsEmpty() throws Exception {
+        assertThat( Strings.isEmpty( null ) ).isTrue();
+        assertThat( Strings.isEmpty( "" ) ).isTrue();
+        assertThat( Strings.isEmpty( " " ) ).isFalse();
+        assertThat( Strings.isEmpty( "not empty" ) ).isFalse();
+    }
+
+    @Test
+    public void testGetSaveVariableNameWithArrayExistingVariables() throws Exception {
+        assertThat( Strings.getSaveVariableName( "int[]" ) ).isEqualTo( "intArray" );
+        assertThat( Strings.getSaveVariableName( "Extends" ) ).isEqualTo( "extends_" );
+        assertThat( Strings.getSaveVariableName( "class" ) ).isEqualTo( "class_" );
+        assertThat( Strings.getSaveVariableName( "Class" ) ).isEqualTo( "class_" );
+        assertThat( Strings.getSaveVariableName( "Case" ) ).isEqualTo( "case_" );
+        assertThat( Strings.getSaveVariableName( "Synchronized" ) ).isEqualTo( "synchronized_" );
+        assertThat( Strings.getSaveVariableName( "prop", "prop", "prop_" ) ).isEqualTo( "prop__" );
+    }
+
+    @Test
+    public void testGetSaveVariableNameWithCollection() throws Exception {
+        assertThat( Strings.getSaveVariableName( "int[]", new ArrayList<String>() ) ).isEqualTo( "intArray" );
+        assertThat( Strings.getSaveVariableName( "Extends", new ArrayList<String>() ) ).isEqualTo( "extends_" );
```

Comment:
I agree with you. I think number prefixes are better on the eye. However, I think that they should be `string1, string2, ..., string10` (camelCase), instead of `string_001` (snake_case).